### PR TITLE
Remove the remaining occurences of DS except for its definition and JParameter.

### PR DESCRIPTION
**Pull request by realityking**
This gets us 90% of the way on removing DS. As soon as we remove JParameter it won't be used by the platform anymore and we can decide to remove it.

### CODE REVIEW

File: tests/suite/joomla/cache/storage/JCacheStorageMock.php [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * @license     GNU General Public License version 2 or later; see LICENSE
  */
 
-JLoader::register('JCacheStorage', JPATH_PLATFORM.DS.'joomla'.DS.'cache'.DS.'storage.php');
+JLoader::register('JCacheStorage', JPATH_PLATFORM . 'joomla/cache/storage.php');
```
Comment (elkuku) [inline tests/suite/joomla/cache/storage/JCacheStorageMock.php]: missing /

Comment (realityking) [inline tests/suite/joomla/cache/storage/JCacheStorageMock.php]: Fixed. Thanks.

File: libraries/joomla/filesystem/folder.php [hunk 2]
```diff
@@ -661,7 +661,6 @@ public static function listFolderTree($path, $filter, $maxLevel = 3, $level = 0,
 	 */
 	public static function makeSafe($path)
 	{
-		//$ds = (DS == '\\') ? '\\/' : DS;
 		$regex = array('#[^A-Za-z0-9:_\\\/-]#');
```
Comment (pasamio) [inline libraries/joomla/filesystem/folder.php:664]: I just realised something, does this mean you can't have folders with periods in them? Like hidden folders?

Comment (piotr-cz) [inline libraries/joomla/filesystem/folder.php:664]: What about non-latin characters (ie. ěščłż), those are permitted in filesystem?
I discourage using them by all means, but I know some people will do it anyway - I have such experience with CMS users who have permission to use media folder.

Comment (realityking) [inline libraries/joomla/filesystem/folder.php:664]: Kinda a different topic :)

**Comment by chdemko** [comment]
The pull request cannot be merged

**Comment by realityking** [comment]
I fixed the conflicts.

**Pull request merged**